c10/core/TensorImpl.cpp

```diff
@@ -780,13 +780,11 @@ c10::intrusive_ptr<TensorImpl> TensorImpl::shallow_copy_and_detach_core(
     VariableVersion&& version_counter,
     bool allow_tensor_metadata_change) const {
   c10::intrusive_ptr<TensorImpl> r;
-  const auto mode_stack_len = c10::impl::TorchDispatchModeTLS::stack_len();
+  const auto maybe_mode = c10::impl::TorchDispatchModeTLS::maybe_highest_mode();
```

Comment:
Adding `maybe_highest_mode` is an easy refactor you could land first

Reply:
Agreed! Will do